frontend/public/components/modals/delete-modal.tsx

```diff
@@ -36,6 +61,20 @@ const DeleteModal = withHandlePromise((props: DeleteModalProps & HandlePromisePr
     props.handlePromise(k8sKill(kind, resource, {}, {}, json), () => {
       props.close();
 
+      if (resource.kind === 'NamespaceStore') {
```

Comment:
I think it needs to be deleted using owner ref

Reply:
i think this is migrated to odf-console.

cc: @rexagod

Reply:
Here it deletes the secret only if the secret is created automatically